--Connected Speakers page to Google Sheets

PrathamGwari opened the pull request:
#### Description
**1. Connected Speakers page to Google Sheets.**
User can now add, remove, update speakers from a google sheets.

**2. Updated Readme**
Added proper steps so that user can connect his google sheet with the speakers page

#### Link to issue this resolves
https://github.com/brittanyjoiner15/eui-event-template/issues/19

#### Video demo link of changes
**https://drive.google.com/file/d/1ykLOjoBvf54J5CEIzpKpKihuMPAZdqn3/view?usp=sharing**

**Assignee - @brittanyjoiner15**
**Assigned to - @prathamgwari**

PrathamGwari committed:
--Connected Speakers page to Google Sheets

brittanyjoiner15 commented on README.md:
```diff
@@ -27,6 +27,76 @@ https://eui-event.vercel.app/
+
+### 1. Create a google sheet
+
+#### 1. Go to https://docs.google.com/ and create a blank sheet
```
Should we make a recommendation for what they should name the sheet?

brittanyjoiner15 replied:
Actually... what if we made a template google sheet, like this : and we just offered for them to copy the sheet and rename it? https://docs.google.com/spreadsheets/d/1OKYzfn9oKqJHJphsipYj_bsPvcjqAjPr2NRjihkKlqg/edit?usp=sharing

PrathamGwari replied:
Noted. I will add a link to the example sheet that I have used in the demo video.

brittanyjoiner15 commented on README.md:
```diff
@@ -27,6 +27,76 @@ https://eui-event.vercel.app/
+#### 1. Go to https://docs.google.com/ and create a blank sheet
+#### 2. Add name, title, team, location, shortBio, pronouns, imageLink headers in the sheet.
+#### 3. Populate the sheet with some data.
+#### 4. Change the sheet name to "Speakers" (Name is case sensitive).
```
I think it's technically the tab name, right?

brittanyjoiner15 replied:
Also, is the `name` field the only one that's case sensitive? Don't all the titles need to be lower case and camelCase if more than one word?

PrathamGwari replied:
Actually by "Name is case sensitive", I meant that the Sheet Name "Speakers" is case sensitive. If you change the sheet name to "speakers" then the code would not work.

I will replace "Name is case sensitive" with "Sheet Name is case sensitive" so that user can understand better.
Also the order of headers (title, team, location, shortBio, pronouns, imageLink) should be same too.

So I will add both of these points.

brittanyjoiner15 replied:
ah that makes sense! yes please call that, thanks!

brittanyjoiner15 commented on README.md:
```diff
@@ -27,6 +27,76 @@ https://eui-event.vercel.app/
+
+#### 1. Go to https://docs.google.com/ and create a blank sheet
+#### 2. Add name, title, team, location, shortBio, pronouns, imageLink headers in the sheet.
+#### 3. Populate the sheet with some data.
```
Can we include a sample of what the data should look like? Something like a schema, for instance: 

`name` : string, 
`imageLink`: string pointing to filepath, or url hosting image

PrathamGwari replied:
Noted.

brittanyjoiner15 commented on README.md:
```diff
@@ -27,6 +27,76 @@ https://eui-event.vercel.app/
+#### 7. Click on settings icon on select type menu and select web app.
+![](https://i.postimg.cc/jjW3jRmj/Deployment-settings.jpg)
+
+#### 8. Add Description.
```
Might want to clarify if they need to change anything in `web app`. Looks like no change is needed, but might want to call that out.

brittanyjoiner15 commented on README.md:
```diff
@@ -27,6 +27,76 @@ https://eui-event.vercel.app/
+
+#### 8. Add Description.
+#### 9. Select Anyone in who has access section and click deploy.
+#### 10. Allow all permission if asked.
```
Probably need to clarify at this point, click `Deploy`, THEN authorize access if asked.

brittanyjoiner15 replied:
Also might want to add that Google might say it's an unverified app and tell you to go `back to safety` and do that whole thing, and maybe explain what it is and that this is okay because it's something you're developing and it's essentially you accessing your own data.

PrathamGwari replied:
Noted.

brittanyjoiner15 reviewed (changes requested):
This looks AWESOME and was so easy!! Just had a couple quick changes that I think it would be worth calling out, but overall this is really close and SO incredibly helpful!! Thank you so much!

PrathamGwari commented:
@brittanyjoiner15 shall I create another pull request with all the changes mentioned above?

PrathamGwari committed:
--Added requested changes

PrathamGwari commented:
@brittanyjoiner15 I have added the requested changes. 

Also about the merge conflicts, I have just changed the class based component to functional component which is working totally fine as seen in the video demo.

Let me know if there are any other changes.

brittanyjoiner15 commented:
@PrathamGwari overall looking good but with the refactor of the speakers panel, we seem to have lost the hover effect on the speaker card (see: https://eui-event.vercel.app/speakers ). Can you update so that's added back in and then fix those merge conflicts? After that should be good to merge! Thanks for the changes here! This is great!

PrathamGwari commented:
> @PrathamGwari overall looking good but with the refactor of the speakers panel, we seem to have lost the hover effect on the speaker card (see: https://eui-event.vercel.app/speakers ). Can you update so that's added back in and then fix those merge conflicts? After that should be good to merge! Thanks for the changes here! This is great!

Sure

brittanyjoiner15 commented:
@PrathamGwari just let me know when that's ready and then ill check one final time and merge it!

brittanyjoiner15 commented on src/components/panels/SpeakersPanel.jsx:
```diff
@@ -7,56 +7,65 @@ import {
 import React from "react";
-import { speakers } from "../../data/speakers";
+import { useEffect } from "react";
+// import { speakers } from "../../data/speakers";
```
Can we remove this comment and instead do something like 
```
// speaker data lives in this spreadsheet: https://docs.google.com/spreadsheets/d/1XgyHXaReTZ3Nq_r7QS18GDvqK_ht010QqnI6PXAnePA/edit#gid=0
```

PrathamGwari replied:
Sure

PrathamGwari committed:
--Resolved merge conflicts

PrathamGwari committed:
--Fixed hover effect

PrathamGwari committed:
--Removed speakers import comment and added sheet link

PrathamGwari commented:
@brittanyjoiner15 I have made the requested changes and also fixed the hover effect.

brittanyjoiner15 reviewed (approved):
This looks great!! Thank you so much for your work on this, it's amazing and will be so useful for teams to be able to have folks organize info in a google sheet and it reflect on the site! Thanks so much!
